packages/oscal-react-library/src/components/OSCALLoader.js

```diff
@@ -81,16 +82,16 @@ export default function OSCALLoader(props) {
     fetch(newOscalUrl)
       .then((response) => {
         if (!response.ok) throw new Error(response.status);
-        else return response.json();
+        else return response.text();
       }, handleError)
       .then((result) => {
-        if (!unmounted.current) {
+        if (!unmounted.current && result) {
+          const oscalObj = Convert.toOscal(result);
```

Comment:
I think you need to have some error handling here to make the error in the promise get caught and handled. Right now when validation fails, it spins forever. 

For example, try loading https://raw.githubusercontent.com/usnistgov/OSCAL/main/json/schema/oscal_assessment-plan_schema.json as a document.

Reply:
Oof -- good idea testing something that is actually invalid. I probably should have considered that code path :D